src/panels/config/backup/dialogs/dialog-restore-backup.ts

```diff
@@ -76,7 +76,12 @@ class DialogRestoreBackup extends LitElement implements HassDialog {
     this._error = undefined;
     this._state = undefined;
     this._stage = undefined;
-    if (this._params.backup.protected) {
+
+    const agentIds = Object.keys(this._params.backup.agents);
+    const preferedAgent = getPreferredAgentForDownload(agentIds);
```

Comment:
Should `getPreferredAgentForDownload` be updated to take into account encryption? It would save CPU cycles to decrypt vs maybe network traffic? Might be a hard one to do automatic though.

Reply:
We already discuss that point with the team and most of the time the unencrypted locations will be the local ones so it will end by the same result.